datastore/mongodb/filters.go

```diff
@@ -141,3 +142,88 @@ func (c *Client) AddFilterDimension(ctx context.Context, fID string, dimension m
 	}
 	return nil
 }
+
+func (c *Client) UpdateFilterDimension(ctx context.Context, filterID string, dimensionName string, dimension model.Dimension, currentETag string) (string, error) {
+	col := c.collections.filters
+
+	lockID, err := col.lock(ctx, filterID)
+	if err != nil {
+		return "", errors.Wrap(err, "error while locking filters collection: %w")
+	}
+
+	defer col.unlock(ctx, lockID)
+
+	filter, err := c.GetFilter(ctx, filterID)
+	if err != nil {
+		return "", &er{
+			err:        errors.Wrapf(err, "unable to fetch filter (%s)", filterID),
+			badRequest: true,
```

Comment:
I think it would be better to follow the exising GetFilter error logic here
```suggestion
		err := &er{
			 errors.Wrapf(err, "unable to fetch filter (%s)", filterID)
		}
		if errors.Is(err, mongodb.ErrNoDocumentFound) {
			err.notFound = true
		}
		return nil, err
```

Reply:
When we refined the story we agreed to stick to Fran's Swagger spec and return a `400` if the filter wasn't found. We can change it (definitely wasn't set in stone), but if we _do_ want to return a `400` then we need to differentiate between a bad request/not found in some way on the Mongo error.

Could be a bool like `failedValidation` rather than `badRequest` in order to make it less of a direct mapping, although we would still need to add an additional error interface/check. 🤔

Let me know what you prefer. I agree that it's not great having response terminology leak into the DB layer, it was just driven by the spec 👍

Reply:
`badRequest` addition to the Mongo package removed, since the validation does indeed handle the `400` response code 👍